### PR TITLE
automatically populate the key field based on the flag name

### PR DESCRIPTION
**Pull request by aaronraff**
Closes: #136 

Whenever the flag name input is updated a check is done to see if
the flag key is either empty or the same as the flag name. If so,
the flag key will be updated accordingly. This check is done to
ensure that custom keys are not overridden when changing the flag
name.

This also moves the flag name field above the flag key field when creating a new flag.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -25,6 +25,8 @@ require (
 	github.com/lib/pq v1.2.0
 	github.com/markphelps/flipt-grpc-go v0.0.0-20190303144529-3ebb133e62c0
 	github.com/mattn/go-sqlite3 v1.11.0
+	github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
```
Comment (aaronraff) [inline go.mod]: @markphelps for whatever reason these always get inserted into the `go.mod` file when I run `make dev` or `make test`, but then go away when I run `go mod tidy`. Do you happen to know any potential causes of this? Is it due to these being used at one time and now they are lingering in the `go.sum` file?

Comment (markphelps) [reply go.mod]: Hmm I'm note sure. What version of go are you running locally? When I pulled down your changes and built them it automatically pruned my `go.mod` file to remove these lines:

```
github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
github.com/modern-go/reflect2 v1.0.1 // indirect
```

**Comment by markphelps** [comment]
This is dope. Thanks @aaronraff! I'm gonna copy this impl to the segment form as well as I forgot to add that as part of the issue.

Also FYI, im working on a [branch/PR](https://github.com/markphelps/flipt/compare/packr) to switch to [gobuffalo/packr](https://github.com/gobuffalo/packr) for embedding the assets, which will get rid of the  `go generate` workflow and should make developing locally a bit easier in the future.

Thanks again!

File: go.mod (hunk 1 above)
Comment (aaronraff) [reply go.mod]: I just realized that I was on 1.12.6. Updated to 1.13.1 and these do not seem to show up anymore. Thanks!

**Pull request merged**